cns/kubecontroller/nodenetworkconfig/conversion_windows.go

```diff
@@ -6,22 +6,29 @@ import (
 
 	"github.com/Azure/azure-container-networking/cns"
 	"github.com/Azure/azure-container-networking/crd/nodenetworkconfig/api/v1alpha"
+	"github.com/pkg/errors"
 )
 
 // createNCRequestFromStaticNCHelper generates a CreateNetworkContainerRequest from a static NetworkContainer.
 // If the NC's DefaultGateway is empty, it will set the 0th IP as the gateway IP and all remaining IPs as
 // secondary IPs. If the gateway is not empty, it will not reserve the 0th IP and add it as a secondary IP.
 //
 //nolint:gocritic //ignore hugeparam
-func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPrefix netip.Prefix, subnet cns.IPSubnet) *cns.CreateNetworkContainerRequest {
+func createNCRequestFromStaticNCHelper(nc v1alpha.NetworkContainer, primaryIPPrefix netip.Prefix, subnet cns.IPSubnet) (*cns.CreateNetworkContainerRequest, error) {
 	secondaryIPConfigs := map[string]cns.SecondaryIPConfig{}
 
 	// if NC DefaultGateway is empty, set the 0th IP to the gateway and add the rest of the IPs
 	// as secondary IPs
 	startingAddr := primaryIPPrefix.Masked().Addr() // the masked address is the 0th IP in the subnet
-	if nc.DefaultGateway == "" {
+	if nc.DefaultGateway == "" && nc.Type == v1alpha.Overlay {
 		nc.DefaultGateway = startingAddr.String()
 		startingAddr = startingAddr.Next()
+	} else if nc.DefaultGateway == "" && nc.Type == v1alpha.VNETBlock {
+		startingAddr = startingAddr.Next()
```

Comment:
Leave a comment on why you are skipping the 0th address block here for primary IP

Reply:
done

Reply:
done